src/frontend/apps/e2e/__tests__/app-impress/language.spec.ts

```diff
@@ -25,4 +25,57 @@ test.describe('Language', () => {
       }),
     ).toBeVisible();
   });
+
+  test('checks the language cookie', async ({ page, context }) => {
+    // check initial user language detected and set as cookie
+    const initial_cookies = await context.cookies();
+    const initial_language_cookie = initial_cookies.find(
+      (cookie) => cookie.name == 'impress_language',
+    );
+    expect(initial_language_cookie?.value).toBe('en');
+
+    // switch user language to fr
+    const header = page.locator('header').first();
+    await header.getByRole('combobox').getByText('English').click();
+    await header.getByRole('option', { name: 'Français' }).click();
+
+    // check if cookie is updated
+    const cookies = await context.cookies();
+    const language_cookie = cookies.find(
+      (cookie) => cookie.name == 'impress_language',
+    );
+    expect(language_cookie?.value).toBe('fr');
+  });
```

Comment:
I think we should remove this test, because it is testing the internal mechanism, and this mechanism could change if we do a refacto. What we want above all, it is that the feature works, `checks the language picker` and `checks that backend uses the language cookie` assert already that.